eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -539,25 +537,66 @@ object Helpers {
+                  Transactions.addSigs(claimDelayed, sig)
+                })
+              }
           }
+        case Transactions.AnchorOutputsCommitmentFormat => Nil
       }
 
       LocalCommitPublished(
         commitTx = tx,
         claimMainDelayedOutputTx = mainDelayedTx.map(_.tx),
-        htlcSuccessTxs = htlcTxes.collect { case c: HtlcSuccessTx => c.tx },
-        htlcTimeoutTxs = htlcTxes.collect { case c: HtlcTimeoutTx => c.tx },
-        claimHtlcDelayedTxs = htlcDelayedTxes.map(_.tx),
+        htlcSuccessTxs = htlcTxs.collect { case c: HtlcSuccessTx => c.tx },
+        htlcTimeoutTxs = htlcTxs.collect { case c: HtlcTimeoutTx => c.tx },
+        claimHtlcDelayedTxs = htlcDelayedTxs.map(_.tx),
         irrevocablySpent = Map.empty)
     }
 
+    /**
+     * Claim the output of a 2nd-stage HTLC transaction and replace the obsolete HTLC transaction in our local commit.
+     */
+    def claimLocalCommitHtlcTxOutput(localCommitPublished: LocalCommitPublished, keyManager: ChannelKeyManager, commitments: Commitments, tx: Transaction, feeEstimator: FeeEstimator, feeTargets: FeeTargets)(implicit log: LoggingAdapter): (LocalCommitPublished, Option[Transaction]) = {
+      import commitments._
+      val isHtlcTx = tx.txIn.map(_.outPoint.txid).contains(localCommitPublished.commitTx.txid) &&
+        tx.txIn.map(_.witness).collect(Scripts.extractPreimageFromHtlcSuccess.orElse(Scripts.extractPaymentHashFromHtlcTimeout)).nonEmpty
+      if (isHtlcTx) {
+        val feeratePerKwDelayed = feeEstimator.getFeeratePerKw(feeTargets.claimMainBlockTarget)
+        val channelKeyPath = keyManager.keyPath(localParams, channelVersion)
+        val localPerCommitmentPoint = keyManager.commitmentPoint(channelKeyPath, commitments.localCommit.index.toInt)
+        val localRevocationPubkey = Generators.revocationPubKey(remoteParams.revocationBasepoint, localPerCommitmentPoint)
+        val localDelayedPubkey = Generators.derivePubKey(keyManager.delayedPaymentPoint(channelKeyPath).publicKey, localPerCommitmentPoint)
+        val htlcDelayedTx = generateTx("claim-htlc-delayed") {
+          Transactions.makeClaimLocalDelayedOutputTx(tx, localParams.dustLimit, localRevocationPubkey, remoteParams.toSelfDelay, localDelayedPubkey, localParams.defaultFinalScriptPubKey, feeratePerKwDelayed).map(claimDelayed => {
+            val sig = keyManager.sign(claimDelayed, keyManager.delayedPaymentPoint(channelKeyPath), localPerCommitmentPoint, TxOwner.Local, commitmentFormat)
+            Transactions.addSigs(claimDelayed, sig)
+          })
+        }
+
+        def updateHtlcTx(newTx: Transaction, previousTxs: List[Transaction]): List[Transaction] = {
+          val replaceAt = previousTxs.indexWhere(_.txIn.head.outPoint == newTx.txIn.head.outPoint)
+          if (replaceAt >= 0) previousTxs.updated(replaceAt, newTx) else previousTxs
+        }
```

Comment:
```suggestion
        def updateHtlcTx(newTx: Transaction, previousTxs: List[Transaction]): List[Transaction] = previousTxs.map {
          case previousTx if previousTx.txIn.head.outPoint == newTx.txIn.head.outPoint => newTx
          case previousTx => previousTx
        }
```

or 

```suggestion
def updateHtlcTx(newTx: Transaction, previousTxs: List[Transaction]): List[Transaction] = previousTxs.map { previousTx =>
          if (previousTx.txIn.head.outPoint == newTx.txIn.head.outPoint) newTx else previousTx
        }
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/1702/commits/5d88187901aa9a1892726540ff68d82dc77a7961